### PR TITLE
docs: add security assessment

### PR DESCRIPTION
**Pull request by SSIRKC**
## Description

As stated during the security assessment the security team is currently moving all Confluence assessment to GitHub.
This assessment includes the documented and agreed version for the portal application and the in scope components.

## Why

The pull adds the assessment to the already existing data flow diagram. The diagram itself has not been changed.

## Issue

None

## Checklist

Please delete options that are not relevant.

- [ x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [ x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [ x] I have created and linked IP issues or requested their creation by a committer
- [ x] I have added copyright and license headers, footers (for .md files) or files (for images)
- [x ] I have performed a self-review of my own code
- [x ] I have successfully tested my changes locally
- [ ] I have added tests that prove my changes work
- [ x] I have checked that new and existing tests pass locally with my changes
- [ x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: developer/Technical Documentation/Architecture/Security-Assessment.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+* UI / UX & Developer Guidelines
+
+### Important Links
+
+- [portal-frontend](https://github.com/eclipse-tractusx/portal-frontend)
+- [portal-frontend-registration](https://github.com/eclipse-tractusx/portal-frontend-registration)
+- [portal-shared-components](https://github.com/eclipse-tractusx/portal-shared-components)
+- [portal-assets](https://github.com/eclipse-tractusx/portal-assets)
+- [portal-backend](https://github.com/eclipse-tractusx/portal-backend)
+
```
Comment (evegufy) [inline developer/Technical Documentation/Architecture/Security-Assessment.md:31]: Could you please also portal-iam https://github.com/eclipse-tractusx/portal-iam?

File: developer/Technical Documentation/Architecture/Security-Assessment.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        NS
+        end
+    PDB[(Portal DB \n Postgres \n EF Core for mapping \n objects to SQL)]
+    end
+    subgraph Keycloak
+      subgraph centralidp
+      K
+      end
+      subgraph sharedidp
+        subgraph Company realms
```
Comment (evegufy) [inline developer/Technical Documentation/Architecture/Security-Assessment.md:79]: Could you please add multiple company realms under sharedidp? Like here https://github.com/SSIRKC/portal-assets/blob/patch-1/developer/Technical%20Documentation/Architecture/Data-Flow.md?plain=1#L54. I'd say at least two, to make clear that there are multiple company realms available in the sharedidp.

**Review by evegufy — CHANGES_REQUESTED** [review]
@SSIRKC Thank you for the PR?

Could you please rebase to the release/v1.7.0 branch? I think you checkout out from the main branch.

Could you please have a more detailed look at the linkages? I think in the version hosted in Confluence we had a lot of dotted links, should that be the case in this diagram as well?

Could you please also add information which other security checks are in place besides the thread modelling? Like the other ones mentioned here https://github.com/eclipse-tractusx/sig-release/issues/107.

File: developer/Technical Documentation/Architecture/Security-Assessment.md (hunk 1 above)
Comment (SSIRKC) [reply developer/Technical Documentation/Architecture/Security-Assessment.md]: Done :)

File: developer/Technical Documentation/Architecture/Security-Assessment.md (hunk 2 above)
Comment (SSIRKC) [reply developer/Technical Documentation/Architecture/Security-Assessment.md]: Hi Evelyn, did I get you right to use the provided diagram? I switched it to the one you provided now.

Comment (evegufy) [reply developer/Technical Documentation/Architecture/Security-Assessment.md]: Hi @SSIRKC , yes using the other diagram is good. That should be the starting point for the following:
> Could you please have a more detailed look at the linkages? I think in the version hosted in Confluence we had a lot of dotted links, should that be the case in this diagram as well?

Comment (evegufy) [reply developer/Technical Documentation/Architecture/Security-Assessment.md]: You could also remove the separate file for the [Data Flow diagram](https://github.com/SSIRKC/portal-assets/blob/patch-1/developer/Technical%20Documentation/Architecture/Data-Flow.md), if it's now completely integrated in the Security Assessment file.

**Comment by evegufy** [comment]
> > @SSIRKC Thank you for the PR?
> > Could you please rebase to the release/v1.7.0 branch? I think you checkout out from the main branch.
> > Could you please have a more detailed look at the linkages? I think in the version hosted in Confluence we had a lot of dotted links, should that be the case in this diagram as well?
> > Could you please also add information which other security checks are in place besides the thread modelling? Like the other ones mentioned here [eclipse-tractusx/sig-release#107](https://github.com/eclipse-tractusx/sig-release/issues/107).
> 
> Hi @evegufy,
> 
> I will correct the branch. For me it looks like branch release/v1.7.0. Have to figure out what happend here.
> 

I don't think the rebase worked. I still the 12 commits added. Could you please check again?

> In Confluence were dotted lines, however, they only serve a visual purpose. I can do this here as well if you like.
> 

Would be great if you could transfer the dotted lines, this might improve the readability of the mermaid diagram, which currently isn't that great.

> The security checks that you normally mention in an assessment are meant to respond to findings within the thread modelling process. I would rather not add SCA/SAST or DAST scan results here, but if you like we could meet up and prepare a list of general controls that we can list here. There are best practices that you can sign off e.g. input validation, encryption of data at rest/transfer/use, etc. Should we do that?

I don't mean to add the scan results, just mentioning which checks we're doing, which purpose they serve and which criteria are we following (for instance no critical or high findings).

**Comment by SSIRKC** [comment]
Added dotted lines and bold lines for better visualization. Also added an overview/legend of symbol meanings. I will also add the security controls.

File: developer/Technical Documentation/Architecture/Security-Assessment.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    MSS <-.-> K
+    MSS ==>|Company app subscription data \n app service data + user preferences| PDB
+    SDF <--> CH
+    SDT -.->|Product meta data| PF
+    NC1 & CU-Shared1 & NC2 & CU-Shared2 & NC3 & CU-Shared3 & CU-Own -.-> |OIDC| K
+    K <-.-> |"Authentication/authorization data (using JWT)"| RF & PF
+    subgraph Legend
+        C1("Component 1")
+        D1[("Database")]
+        C1 <-.-> D1
```
Comment (evegufy) [inline developer/Technical Documentation/Architecture/Security-Assessment.md:143]: Please also add description of this link like done for <--> and https connections.

File: developer/Technical Documentation/Architecture/Security-Assessment.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        C1 <--> |https connections| D1 
+        C1 <==> D1
```
Comment (evegufy) [inline developer/Technical Documentation/Architecture/Security-Assessment.md:145]: Please also add description of this link like done for <--> and https connections.

**Comment by evegufy** [comment]
@SSIRKC could you please give me an estimate until when you'll be able to add the last changes? I need to close the v1.7.0 latest by tomorrow.

File: developer/Technical Documentation/Architecture/Security-Assessment.md (hunk 3 above)
Comment (SSIRKC) [reply developer/Technical Documentation/Architecture/Security-Assessment.md]: Done :)

File: developer/Technical Documentation/Architecture/Security-Assessment.md (hunk 4 above)
Comment (SSIRKC) [reply developer/Technical Documentation/Architecture/Security-Assessment.md]: Done :)

**Review by evegufy — APPROVED** [review]
@SSIRKC Thanks you! Looks good.
I just did some small changes still.

**Pull request merged**